packages/web_benchmarks/lib/src/browser.dart

```diff
@@ -0,0 +1,599 @@
+        '--user-data-dir=${options.userDataDirectory}',
+      if (options.url != null) options.url,
+      if (io.Platform.environment['CHROME_NO_SANDBOX'] == 'true')
+        '--no-sandbox',
+      if (options.headless) '--headless',
+      if (withDebugging) '--remote-debugging-port=${options.debugPort}',
+      '--window-size=${options.windowWidth},${options.windowHeight}',
+      '--disable-extensions',
+      '--disable-popup-blocking',
+      // Indicates that the browser is in "browse without sign-in" (Guest session) mode.
+      '--bwsi',
+      '--no-first-run',
+      '--no-default-browser-check',
+      '--disable-default-apps',
+      '--disable-translate',
+    ];
+    final io.Process chromeProcess = await io.Process.start(
+      _findSystemChromeExecutable(),
+      args,
+      workingDirectory: workingDirectory,
+    );
+
+    WipConnection debugConnection;
+    if (withDebugging) {
+      debugConnection =
+          await _connectToChromeDebugPort(chromeProcess, options.debugPort);
+    }
+
+    return Chrome._(chromeProcess, onError, debugConnection);
+  }
+
+  final io.Process _chromeProcess;
+  final ChromeErrorCallback _onError;
+  final WipConnection _debugConnection;
+  bool _isStopped = false;
+
+  Completer<void> _tracingCompleter;
+  StreamSubscription<WipEvent> _tracingSubscription;
+  List<Map<String, dynamic>> _tracingData;
+
+  /// Starts recording a performance trace.
+  ///
+  /// If there is already a tracing session in progress, throws an error. Call
+  /// [endRecordingPerformance] before starting a new tracing session.
+  ///
+  /// The [label] is for debugging convenience.
+  Future<void> beginRecordingPerformance(String label) async {
+    if (_tracingCompleter != null) {
+      throw StateError(
+          'Cannot start a new performance trace. A tracing session labeled '
+          '"$label" is already in progress.');
+    }
+    _tracingCompleter = Completer<void>();
+    _tracingData = <Map<String, dynamic>>[];
+
+    // Subscribe to tracing events prior to calling "Tracing.start". Otherwise,
+    // we'll miss tracing data.
+    _tracingSubscription =
+        _debugConnection.onNotification.listen((WipEvent event) {
+      // We receive data as a sequence of "Tracing.dataCollected" followed by
+      // "Tracing.tracingComplete" at the end. Until "Tracing.tracingComplete"
+      // is received, the data may be incomplete.
+      if (event.method == 'Tracing.tracingComplete') {
+        _tracingCompleter.complete();
+        _tracingSubscription.cancel();
+        _tracingSubscription = null;
+      } else if (event.method == 'Tracing.dataCollected') {
+        final dynamic value = event.params['value'];
+        if (value is! List) {
+          throw FormatException(
+              '"Tracing.dataCollected" returned malformed data. '
+              'Expected a List but got: ${value.runtimeType}');
+        }
+        _tracingData.addAll(event.params['value'].cast<Map<String, dynamic>>());
```

Comment:
`event.params['value']` seems to be already of type `Map<String, dynamic>`. Why do we need to cast here?

Reply:
While it's true that the list _contains_ `Map<String, dynamic>` the reified type of the list is `List<dynamic>`, so we have to cast the element type of the list from `dynamic` to `Map<String, dynamic>`.